### PR TITLE
More changes for Python3 compatibility

### PR DESCRIPTION
**Pull request by marmeladema**
Following #24435 for #23607 here are even more changes to be compatible with Python3.
I managed to get `./mach build` properly work with Python3 but `test-tidy` does not work yet because of a lot of problems in `web-platform-tests` which i will have to deal with at some point.

Because flake8 appears to be incompatible with the distro package in some way, i had to change to way we call flake8 (which was deprecated anyway). With this change, we should be able to update flake8 to a recent version but subprocess can be surprising on Windows platform so i'd like someone to try out those changes.

---

- [x] `./mach build -d` does not report any errors
- [x] `./mach test-tidy` does not report any errors

### CODE REVIEW

**Comment by marmeladema** [comment]
@jdm i don't know how to add you as reviewer so i thought i would just ping you :+1:

**Review by JacobAWilkins — APPROVED** [review]
Very nice

**Comment by jdm** [comment]
r? @SimonSapin
Honestly ai think Simon has more experience reviewing python3 than I do.

**Comment by marmeladema** [comment]
> r? @SimonSapin
> Honestly I think Simon has more experience reviewing python3 than I do.

Yeah i did not imply to remove @SimonSapin from reviewers, just wanted to add you

File: python/tidy/servo_tidy/tidy.py
Comment (SimonSapin) [inline python/tidy/servo_tidy/tidy.py]: Why create this temporary file rather than pass `file_name` to flake8?

**Review by SimonSapin — COMMENTED** [review]
I’d like to also get some testing so we don’t accidentally regress this. If `./mach build` is supposed to work on Python 3 now, please change it to `python3 mach build` here, for example:

https://github.com/servo/servo/blob/5fb941d6f7de9289b5ffa683c924c2e2bba74d3a/etc/taskcluster/decision_task.py#L180

File: python/tidy/servo_tidy/tidy.py
Comment (marmeladema) [reply python/tidy/servo_tidy/tidy.py]: I wondered that myself and choose to dump 
the content because I did not take time to figure out if file_name really existed on disk

**Comment by marmeladema** [comment]
> I’d like to also get some testing so we don’t accidentally regress this. If `./mach build` is supposed to work on Python 3 now, please change it to `python3 mach build` here, for example:
> 
> https://github.com/servo/servo/blob/5fb941d6f7de9289b5ffa683c924c2e2bba74d3a/etc/taskcluster/decision_task.py#L180

It passes on a branch based on this one with a few more tweaks. We can either merge this one, and i'll come back in a few days with the everything sorted out to build with python3 or i add the remaining bits in this one.

Also about just changing `./mach build` to `python3 mach build`, i wonder the effect on the subsequent commands that will run with python2 like `./mach test-tidy --no-progress --self-test`

**Comment by SimonSapin** [comment]
I’ll leave it up to you whether you prefer to add a commit in this PR for those few more tweaks (maybe they’re already ready) or if you prefer not to block landing this PR and make a new one later.

I don’t expect that a Python 3 command followed by a Python 2 command would cause any trouble, they’re independent processes. Finding out for sure is what CI is for!

**Comment by marmeladema** [comment]
> I’ll leave it up to you whether you prefer to add a commit in this PR for those few more tweaks (maybe they’re already ready) or if you prefer not to block landing this PR and make a new one later.

Then i would prefer merging this now and come back later with the remaining bits. My other branch actually only check that you have python3, not python2 anymore (even though the code remains compatible) and will take some time to make python version detection compatible as its not ready yet.

> I don’t expect that a Python 3 command followed by a Python 2 command would cause any trouble, they’re independent processes. Finding out for sure is what CI is for!

We might have to remove `python/_virtualenv` between python3 and python2 though

File: python/tidy/servo_tidy/tidy.py
Comment (marmeladema) [reply python/tidy/servo_tidy/tidy.py]: Should I change that before merging?

Comment (SimonSapin) [reply python/tidy/servo_tidy/tidy.py]: Yes, please do.

**Comment by SimonSapin** [comment]
Oh yeah, there’s virtualenv. It may be worth having it at a different location based on the Python version.

File: python/tidy/servo_tidy/tidy.py
Comment (marmeladema) [reply python/tidy/servo_tidy/tidy.py]: Done!

**Comment by SimonSapin** [comment]
Looks good, thanks!

@bors-servo r+

**Comment by CYBAI** [comment]
@bors-servo retry

- network issue in magicleap build

**Comment by marmeladema** [comment]
> @bors-servo retry
> 
>     * network issue in magicleap build

I don't think the `retry` worked?

**Comment by jdm** [comment]
It does work; there is a queue at https://build.servo.org/homu/queue/servo.

**Pull request merged**
